src/tasks/tasks.hpp

```diff
@@ -544,6 +544,8 @@ class TaskRegion {
 
  public:
   TaskRegion() = delete;
+  TaskRegion(const TaskRegion &) = delete; // Prevent copying TaskRegions during AddRegion
+                                           // calls which is a segfault
```

Comment:
I ran into a nasty bug where if you mistakenly do
```c++
TaskRegion my_region = tc.AddRegion(N);
```
rather than storing the reference
```c++
TaskRegion &my_region = tc.AddRegion(N);
```
you get a very unintuitive segfault. I think we always want to return and store the reference from this function -- any reason not to delete the copy constructor to enforce this?